spark3/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -159,6 +160,10 @@ public StructType schema() {
 
   @Override
   public ScanBuilder newScanBuilder(CaseInsensitiveStringMap options) {
+    if (options.containsKey(SparkReadOptions.FILE_SCAN_TASK_SET_ID)) {
```

Comment:
It would be nice to have a comment explaining this.

Reply:
Added a comment.